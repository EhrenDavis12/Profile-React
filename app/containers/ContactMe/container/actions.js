/*
 *
 * ContactMePage actions
 *
 */

import {
  SUBMIT_MESSAGE_FORM,
  SUBMIT_MESSAGE_FORM_SUCCEEDED,
  SUBMIT_MESSAGE_FORM_FAILED,
  CLOSED_SUCCESS_MESSAGE,
} from './constants';

export function submitMessageForm(messageForm) {
  return {
    type: SUBMIT_MESSAGE_FORM,
    messageForm,
  };
}

export function submitMessageFormSucceeded() {
  return {
    type: SUBMIT_MESSAGE_FORM_SUCCEEDED,
  };
}

export function submitMessageFormFailed(message) {
  return {
    type: SUBMIT_MESSAGE_FORM_FAILED,
    message,
  };
}

export function closedSuccessMessage() {
  return {
    type: CLOSED_SUCCESS_MESSAGE,
  };
}
