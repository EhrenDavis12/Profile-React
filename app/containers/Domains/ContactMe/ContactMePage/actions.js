/*
 *
 * ContactMePage actions
 *
 */

import {
  SUBMIT_MESSAGE_FORM,
  SUBMIT_MESSAGE_FORM_SUCCEEDED,
  SUBMIT_MESSAGE_FORM_FAILED,
} from './constants';

export function submitMessageForm(messageForm) {
  return {
    type: SUBMIT_MESSAGE_FORM,
    messageForm,
  };
}

export function submitMessageFormSucceeded() {
  return {
    type: SUBMIT_MESSAGE_FORM_SUCCEEDED,
  };
}

export function submitMessageFormFailed(message) {
  return {
    type: SUBMIT_MESSAGE_FORM_FAILED,
    message,
  };
}
