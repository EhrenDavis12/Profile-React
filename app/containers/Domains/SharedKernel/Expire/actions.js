/*
 *
 * Expire actions
 *
 */

import {
  CLEAR_TIMER_ACTION,
  SET_TIMER_ACTION,
  SHOW_CHILDREN_ACTION,
  HIDE_CHILDREN_ACTION,
  SET_TIMER_SUCCEEDED,
  CLEAR_TIMER_SUCCEEDED,
} from './constants';

export function setTimerAction(
  delay,
  onHideChildrenAction,
  onClearTimerAction,
) {
  return {
    type: SET_TIMER_ACTION,
    delay,
    onHideChildrenAction,
    onClearTimerAction,
  };
}

export function setTimerSucceeded(timer) {
  return {
    type: SET_TIMER_SUCCEEDED,
    timer,
  };
}

export function clearTimerAction() {
  return {
    type: CLEAR_TIMER_ACTION,
  };
}

export function clearTimerSucceeded() {
  return {
    type: CLEAR_TIMER_SUCCEEDED,
  };
}

export function showChildrenAction() {
  return {
    type: SHOW_CHILDREN_ACTION,
  };
}

export function hideChildrenAction() {
  return {
    type: HIDE_CHILDREN_ACTION,
  };
}
