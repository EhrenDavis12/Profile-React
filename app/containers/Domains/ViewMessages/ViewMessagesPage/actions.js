/*
 *
 * ViewMessagesPage actions
 *
 */

import {
  REQUEST_USER,
  REQUEST_USER_SUCCEEDED,
  REQUEST_USER_FAILED,
  REQUEST_USER_MESSAGES,
  REQUEST_USER_MESSAGES_SUCCEEDED,
  REQUEST_USER_MESSAGES_FAILED,
  SELECT_USER_MESSAGE,
} from './constants';

export function requestUser() {
  return {
    type: REQUEST_USER,
  };
}

export function requestUserSucceeded(user) {
  return {
    type: REQUEST_USER_SUCCEEDED,
    user,
  };
}

export function requestUserFailed(message) {
  return {
    type: REQUEST_USER_FAILED,
    message,
  };
}

export function requestUserMessages() {
  return {
    type: REQUEST_USER_MESSAGES,
  };
}

export function requestUserMessagesSucceeded(userMessages) {
  return {
    type: REQUEST_USER_MESSAGES_SUCCEEDED,
    userMessages,
  };
}

export function requestUserMessagesFailed(message) {
  return {
    type: REQUEST_USER_MESSAGES_FAILED,
    message,
  };
}

export function selectUserMessage(selectedUserMessage) {
  return {
    type: SELECT_USER_MESSAGE,
    selectedUserMessage,
  };
}
