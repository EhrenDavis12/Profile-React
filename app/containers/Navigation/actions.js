/*
 *
 * Navigation actions
 *
 */

import { TOGGLE_DRAWER, SELECT_LINK } from './constants';

export function toggleDrawer() {
  return {
    type: TOGGLE_DRAWER,
  };
}

export function selectLink(selectedLink) {
  return {
    type: SELECT_LINK,
    selectedLink,
  };
}
