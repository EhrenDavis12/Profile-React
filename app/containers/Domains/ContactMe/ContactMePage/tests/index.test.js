import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import 'jest-styled-components';
import configureStore from '../../../../../configureStore';
// import { enzymeFind } from 'styled-components/test-utils';

// import { ContactMePage } from '../index';
import { submitMessageForm, closedSuccessMessage } from '../actions';
import { ContactMePage, mapDispatchToProps } from '../index';

describe('<ContactMePage />', () => {
  let store;
  let wrapper;
  let submitMessageFormSpy;
  const renderComponent = _messageStatus =>
    mount(
      <Provider store={store}>
        <ContactMePage
          messageStatus={_messageStatus}
          submitMessageForm={submitMessageFormSpy}
        />
      </Provider>,
    );

  beforeAll(() => {
    submitMessageFormSpy = jest.fn();
    store = configureStore({}, browserHistory);
  });
  describe('<ContactMePage />', () => {
    it('should render its UserMessageBody succeeded', () => {
      wrapper = renderComponent('succeeded');
      expect(wrapper).toBeDefined();
      expect(wrapper.find('Navigation').length).toEqual(1);
      expect(wrapper.find('HeaderBar').length).toEqual(1);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render its UserMessageBody closed', () => {
      wrapper = renderComponent('closed');
      expect(wrapper).toBeDefined();
      expect(wrapper.find('Navigation').length).toEqual(1);
      expect(wrapper.find('HeaderBar').length).toEqual(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('mapDispatchToProps', () => {
  describe('submitMessageForm', () => {
    it('should be injected', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      expect(result.submitMessageForm).toBeDefined();
    });

    it('should dispatch requestUser when called', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      result.submitMessageForm();
      expect(dispatch).toHaveBeenCalledWith(submitMessageForm());
    });
  });

  describe('actionAfterExpire', () => {
    it('should be injected', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      expect(result.actionAfterExpire).toBeDefined();
    });

    it('should dispatch actionAfterExpire when called', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      result.actionAfterExpire('Test');
      expect(dispatch).toHaveBeenCalledWith(closedSuccessMessage('Test'));
    });
  });
});
