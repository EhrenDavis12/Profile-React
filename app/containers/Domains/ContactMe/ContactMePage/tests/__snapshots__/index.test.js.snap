// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactMePage /> <ContactMePage /> should render its UserMessageBody closed 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {
        "Navigation": [Function],
      },
      "injectedSagas": Object {
        "Navigation": Object {
          "mode": "@@saga-injector/restart-on-remount",
          "saga": [Function],
          "task": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": undefined,
            "done": Promise {},
            "error": [Function],
            "id": 6,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRunning": [Function],
            "joiners": Array [],
            "name": "rootSaga",
            "result": [Function],
            "setContext": [Function],
          },
        },
      },
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <ContactMePage
    messageStatus="closed"
    submitMessageForm={[MockFunction]}
  >
    <withReducer(withSaga(Connect(Navigation)))>
      <withSaga(Connect(Navigation))>
        <Connect(Navigation)>
          <Navigation
            isDrawerOpen={false}
            navigationLinks={
              Array [
                Object {
                  "DisplayName": "About Me",
                  "LinkTo": "/",
                  "key": "AboutMe",
                },
                Object {
                  "DisplayName": "Portfolio",
                  "LinkTo": "/portfolio",
                  "key": "Portfolio",
                },
                Object {
                  "DisplayName": "Contact",
                  "LinkTo": "/contact",
                  "key": "Contact",
                },
                Object {
                  "DisplayName": "Messages",
                  "LinkTo": "/messages",
                  "key": "Messages",
                },
              ]
            }
            selectItem={[Function]}
            toggleDrawer={[Function]}
          >
            <div>
              <NavigationBody
                isDrawerOpen={false}
                navigationLinks={
                  Array [
                    Object {
                      "DisplayName": "About Me",
                      "LinkTo": "/",
                      "key": "AboutMe",
                    },
                    Object {
                      "DisplayName": "Portfolio",
                      "LinkTo": "/portfolio",
                      "key": "Portfolio",
                    },
                    Object {
                      "DisplayName": "Contact",
                      "LinkTo": "/contact",
                      "key": "Contact",
                    },
                    Object {
                      "DisplayName": "Messages",
                      "LinkTo": "/messages",
                      "key": "Messages",
                    },
                  ]
                }
                selectItem={[Function]}
                toggleDrawer={[Function]}
              >
                <AppBar
                  toggleDrawer={[Function]}
                >
                  <div
                    className="appBar"
                  >
                    <BasicButton
                      className="iconButton"
                      onClick={[Function]}
                    >
                      <div
                        className="iconButton"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        <FontAwesome
                          className="icon"
                          key=".0"
                          name="bars"
                        >
                          <span
                            aria-hidden={true}
                            className="fa fa-bars icon"
                          />
                        </FontAwesome>
                      </div>
                    </BasicButton>
                    <div
                      className="heading"
                    >
                      Coder Daily
                    </div>
                    <div
                      className="linkContainer"
                    >
                      Log In
                    </div>
                  </div>
                </AppBar>
                <Drawer
                  isDrawerOpen={false}
                  itemKeyAttr="key"
                  itemLabelAttr="DisplayName"
                  items={
                    Array [
                      Object {
                        "DisplayName": "About Me",
                        "LinkTo": "/",
                        "key": "AboutMe",
                      },
                      Object {
                        "DisplayName": "Portfolio",
                        "LinkTo": "/portfolio",
                        "key": "Portfolio",
                      },
                      Object {
                        "DisplayName": "Contact",
                        "LinkTo": "/contact",
                        "key": "Contact",
                      },
                      Object {
                        "DisplayName": "Messages",
                        "LinkTo": "/messages",
                        "key": "Messages",
                      },
                    ]
                  }
                  selectItem={[Function]}
                >
                  <div
                    className="drawer"
                  >
                    <BasicButton
                      className="item"
                      key="AboutMe"
                      onClick={[Function]}
                    >
                      <div
                        className="item"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        About Me
                      </div>
                    </BasicButton>
                    <BasicButton
                      className="item"
                      key="Portfolio"
                      onClick={[Function]}
                    >
                      <div
                        className="item"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        Portfolio
                      </div>
                    </BasicButton>
                    <BasicButton
                      className="item"
                      key="Contact"
                      onClick={[Function]}
                    >
                      <div
                        className="item"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        Contact
                      </div>
                    </BasicButton>
                    <BasicButton
                      className="item"
                      key="Messages"
                      onClick={[Function]}
                    >
                      <div
                        className="item"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        Messages
                      </div>
                    </BasicButton>
                  </div>
                </Drawer>
              </NavigationBody>
            </div>
          </Navigation>
        </Connect(Navigation)>
      </withSaga(Connect(Navigation))>
    </withReducer(withSaga(Connect(Navigation)))>
    <HeaderBar
      header="Contact Me"
      subMessage="Feel free to send me a message and i will get back to you"
    >
      <div
        className="jumbotron"
      >
        <h1>
          Contact Me
        </h1>
        <p>
          Feel free to send me a message and i will get back to you
        </p>
      </div>
    </HeaderBar>
    <ContactBody
      messageStatus="closed"
      submitMessageForm={[MockFunction]}
    >
      <div
        className="messageForm"
      >
        <input
          className="input"
          id="emailInput"
          placeholder="Your Email"
          type="email"
        />
        <input
          className="input"
          id="subjectInput"
          placeholder="Subject"
          type="text"
        />
        <textarea
          className="inputLargeText"
          id="messageInput"
          placeholder="Your Message"
          type="text"
        />
        <div
          className="actionContainer"
        >
          <BasicButton
            className="button"
            id="clear"
            onClick={[Function]}
          >
            <div
              className="button"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Clear
            </div>
          </BasicButton>
          <BasicButton
            className="button"
            id="submit"
            onClick={[Function]}
          >
            <div
              className="button"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Submit
            </div>
          </BasicButton>
        </div>
      </div>
    </ContactBody>
  </ContactMePage>
</Provider>
`;

exports[`<ContactMePage /> <ContactMePage /> should render its UserMessageBody succeeded 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {
        "Navigation": [Function],
      },
      "injectedSagas": Object {
        "Navigation": Object {
          "mode": "@@saga-injector/restart-on-remount",
          "saga": [Function],
          "task": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": undefined,
            "done": Promise {},
            "error": [Function],
            "id": 1,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRunning": [Function],
            "joiners": Array [],
            "name": "rootSaga",
            "result": [Function],
            "setContext": [Function],
          },
        },
      },
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <ContactMePage
    messageStatus="succeeded"
    submitMessageForm={[MockFunction]}
  >
    <withReducer(withSaga(Connect(Navigation)))>
      <withSaga(Connect(Navigation))>
        <Connect(Navigation)>
          <Navigation
            isDrawerOpen={false}
            navigationLinks={
              Array [
                Object {
                  "DisplayName": "About Me",
                  "LinkTo": "/",
                  "key": "AboutMe",
                },
                Object {
                  "DisplayName": "Portfolio",
                  "LinkTo": "/portfolio",
                  "key": "Portfolio",
                },
                Object {
                  "DisplayName": "Contact",
                  "LinkTo": "/contact",
                  "key": "Contact",
                },
                Object {
                  "DisplayName": "Messages",
                  "LinkTo": "/messages",
                  "key": "Messages",
                },
              ]
            }
            selectItem={[Function]}
            toggleDrawer={[Function]}
          >
            <div>
              <NavigationBody
                isDrawerOpen={false}
                navigationLinks={
                  Array [
                    Object {
                      "DisplayName": "About Me",
                      "LinkTo": "/",
                      "key": "AboutMe",
                    },
                    Object {
                      "DisplayName": "Portfolio",
                      "LinkTo": "/portfolio",
                      "key": "Portfolio",
                    },
                    Object {
                      "DisplayName": "Contact",
                      "LinkTo": "/contact",
                      "key": "Contact",
                    },
                    Object {
                      "DisplayName": "Messages",
                      "LinkTo": "/messages",
                      "key": "Messages",
                    },
                  ]
                }
                selectItem={[Function]}
                toggleDrawer={[Function]}
              >
                <AppBar
                  toggleDrawer={[Function]}
                >
                  <div
                    className="appBar"
                  >
                    <BasicButton
                      className="iconButton"
                      onClick={[Function]}
                    >
                      <div
                        className="iconButton"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        <FontAwesome
                          className="icon"
                          key=".0"
                          name="bars"
                        >
                          <span
                            aria-hidden={true}
                            className="fa fa-bars icon"
                          />
                        </FontAwesome>
                      </div>
                    </BasicButton>
                    <div
                      className="heading"
                    >
                      Coder Daily
                    </div>
                    <div
                      className="linkContainer"
                    >
                      Log In
                    </div>
                  </div>
                </AppBar>
                <Drawer
                  isDrawerOpen={false}
                  itemKeyAttr="key"
                  itemLabelAttr="DisplayName"
                  items={
                    Array [
                      Object {
                        "DisplayName": "About Me",
                        "LinkTo": "/",
                        "key": "AboutMe",
                      },
                      Object {
                        "DisplayName": "Portfolio",
                        "LinkTo": "/portfolio",
                        "key": "Portfolio",
                      },
                      Object {
                        "DisplayName": "Contact",
                        "LinkTo": "/contact",
                        "key": "Contact",
                      },
                      Object {
                        "DisplayName": "Messages",
                        "LinkTo": "/messages",
                        "key": "Messages",
                      },
                    ]
                  }
                  selectItem={[Function]}
                >
                  <div
                    className="drawer"
                  >
                    <BasicButton
                      className="item"
                      key="AboutMe"
                      onClick={[Function]}
                    >
                      <div
                        className="item"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        About Me
                      </div>
                    </BasicButton>
                    <BasicButton
                      className="item"
                      key="Portfolio"
                      onClick={[Function]}
                    >
                      <div
                        className="item"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        Portfolio
                      </div>
                    </BasicButton>
                    <BasicButton
                      className="item"
                      key="Contact"
                      onClick={[Function]}
                    >
                      <div
                        className="item"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        Contact
                      </div>
                    </BasicButton>
                    <BasicButton
                      className="item"
                      key="Messages"
                      onClick={[Function]}
                    >
                      <div
                        className="item"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        Messages
                      </div>
                    </BasicButton>
                  </div>
                </Drawer>
              </NavigationBody>
            </div>
          </Navigation>
        </Connect(Navigation)>
      </withSaga(Connect(Navigation))>
    </withReducer(withSaga(Connect(Navigation)))>
    <HeaderBar
      header="Contact Me"
      subMessage="Feel free to send me a message and i will get back to you"
    >
      <div
        className="jumbotron"
      >
        <h1>
          Contact Me
        </h1>
        <p>
          Feel free to send me a message and i will get back to you
        </p>
      </div>
    </HeaderBar>
    <Expire
      delay={1500}
      messageStatus="succeeded"
      submitMessageForm={[MockFunction]}
    >
      <div>
        <ResultBanner
          message="Message Sent"
          resultClass="succeeded"
        >
          <div
            className="grayOut"
          >
            <div
              className="succeeded"
            >
              Message Sent
            </div>
          </div>
        </ResultBanner>
      </div>
    </Expire>
    <ContactBody
      messageStatus="succeeded"
      submitMessageForm={[MockFunction]}
    >
      <div
        className="messageForm"
      >
        <input
          className="input"
          id="emailInput"
          placeholder="Your Email"
          type="email"
        />
        <input
          className="input"
          id="subjectInput"
          placeholder="Subject"
          type="text"
        />
        <textarea
          className="inputLargeText"
          id="messageInput"
          placeholder="Your Message"
          type="text"
        />
        <div
          className="actionContainer"
        >
          <BasicButton
            className="button"
            id="clear"
            onClick={[Function]}
          >
            <div
              className="button"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Clear
            </div>
          </BasicButton>
          <BasicButton
            className="button"
            id="submit"
            onClick={[Function]}
          >
            <div
              className="button"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Submit
            </div>
          </BasicButton>
        </div>
      </div>
    </ContactBody>
  </ContactMePage>
</Provider>
`;
